src/EditorFeatures/CSharpTest/CodeActions/InvertIf/InvertIfTests.Elseless.cs

```diff
@@ -0,0 +1,602 @@
+        {
+            [||]if (!c)
+            {
+                return;
+            }
+            break;
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        foreach (var item in list)
+        {
+            [||]if (c)
+            {
+                break;
+            }
+            return;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestElseless06()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        foreach (var item in list)
+        {
+            {
+                [||]if (c)
+                {
+                    f();
+                }
+            }
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        foreach (var item in list)
+        {
+            {
+                [||]if (!c)
+                {
+                    continue;
+                }
+
+                f();
+            }
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestElseless07()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        [||]if (c)
+        {
+            f();
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        [||]if (!c)
+        {
+            return;
+        }
+
+        f();
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestElseless08()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        switch (o)
+        {
+            case 1:
+                [||]if (c)
+                {
+                    f();
+                    f();
+                }
+
+                break;
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        switch (o)
+        {
+            case 1:
+                [||]if (!c)
+                {
+                    break;
+                }
+
+                f();
+                f();
+
+                break;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestElseless09()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        switch (o)
+        {
+            case 1:
+                [||]if (c)
+                {
+                    if (c)
+                    {
+                        return 1;
+                    }
+                }
+
+                return 2;
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        switch (o)
+        {
+            case 1:
+                [||]if (!c)
+                {
+                    return 2;
+                }
+
+                if (c)
+                {
+                    return 1;
+                }
+
+                return 2;
```

Comment:
this is a bit freaky.  Are you straight up duplicating the code?   I'm not sure how i feel aobut this. :-/

Reply:
This is the most compact replacement in this particular situation, comes in handy most of the time (e.g. returning a constant). otherwise you can just undo :) (R# do this as well.)

Reply:
(bydesign)